OG/OG/OG.g4

```diff
@@ -1,83 +1,216 @@
+
+toCommands: toCmd=toCommand chainedToCmds=toCommands          #chainedToCommand
+          | toCmd=toCommand                                   #singleToCommand
+          ;
+
+   
+curveCommand    : 'curve''.''withAngle' '('angle=mathExpression ')'  fromCmd=fromCommand toCmd=toCommand;
+                
+toCommand       : '.to''(' id=ID ')'                   #toWithId
+                | '.to''(' tuple=numberTuple ')'          #toWithNumberTuple
+                | '.to''(' toPoint=StartPointReference ')'  #toWithStartPointRef
+                | '.to''(' toPoint=EndPointReference ')'    #toWithEndPointRef
+                ;
+
+fromCommand     :  '.from' '(' id=ID')'                             #fromWithId
+                |  '.from' '(' tuple=numberTuple ')'                #fromWithNumberTuple
+                |  '.from' '(' fromPoint=StartPointReference ')'    #fromWithStartPointRef
+                |  '.from' '(' fromPoint=EndPointReference ')'      #fromWithEndPointRef
+                ;
 
 
 //Iterative statements:
-iterationCommand: numberIteration | untilIteration;
-numberIteration : 'repeat''('mathExpression')' body 'repeat.end'; //Ret krop?
-untilIteration  : 'repeat''.''until''('(boolExpression | functionCall)')' body 'repeat.end';
+iterationCommand: numberIterCmd=numberIteration 
+                | untilIterCmd=untilIteration
+                ;
+numberIteration : 'repeat''('iterator=mathExpression')' statements=body 'repeat.end'
+                ;
+untilIteration  : 'repeat''.''until''(' iterator=functionCall')' statements=body 'repeat.end'    #untilFuncCall
+                | 'repeat''.''until''(' iterator=boolExpression')' statements=body 'repeat.end'  #untilCondition
+                ; //Udvid til flere regler?
 
 
 //Functions: 
-functionDeclaration     : returnFunctionDCL | voidFunctionDCL;
-returnFunctionDCL       : 'function' typeWord ID parameterDeclarations '{' body returnStatement '}';
-typeWord                : (PointDCLWord | BoolDCLWord | NumberDCLWord);
-voidFunctionDCL         : 'function' 'void' ID parameterDeclarations '{' body '}';
-
-parameterDeclarations   : '('parameters')';
-parameters              :  (typeWord ID',')* typeWord ID | ;
-functionCall            : ID '(' parameterList ')';
-parameterList           : ( (ID | expression | CoordinateXYValue)',')* (ID | expression | CoordinateXYValue) | ;
-returnStatement         : 'return' (ID | expression) ';';
+functionDcl             : returnFunctionDCL 
+                        | voidFunctionDCL
+                        ;
+                        
+returnFunctionDCL       : 'function' type=typeWord funcName=ID '('paramDcls=parameterDeclarations ')''{' statements=body returnStmt=returnStatement '}';
+
+typeWord                : PointDCLWord 
+                        | BoolDCLWord 
+                        | NumberDCLWord
+                        ;
+voidFunctionDCL         : 'function' 'void' id=ID '(' paramDcls=parameterDeclarations ')'  '{' statements=body '}';
+
```

Comment:
Har vi behov for void funcs? Eller er vores shapes i virkeligheden vores "void funktioner"?
Hvad skulle de bruges til? :)

(Plus det er liiiidt c agtigt, men det går nok xD)